### PR TITLE
Check types with mypy

### PR DESCRIPTION
**Pull request by eide**
- Run each pyX.Y tox run as different actions
 - Fixed small type issues found by mypy

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ format:
 
 .PHONY: test
 test:
-	python3 -m tox -p auto
+	python3 -m tox -p auto -o
```
Comment (simenheg) [inline Makefile:16]: The output gets very messy when run in parallel. I think we should turn off the `-p` option if we deem readable test output more valuable than those few extra seconds it takes to run (I do).

**Review by simenheg — COMMENTED** [review]
Very nice to get mypy into the test matrix!

Just a question regarding test output readability.

File: Makefile (hunk 1 above)
Comment (eide) [reply Makefile]: Removed parallel testing.

**Review by simenheg — DISMISSED** [review]

File: Makefile (hunk 1 above)
Comment (fredriv) [reply Makefile]: `tox` still runs tests in parallel, the `-o` flag just ensures that the output is more suitable to a CI/CD environment (e.g. no spinner): https://tox.readthedocs.io/en/latest/example/basic.html#parallel-mode

File: Makefile [hunk 2]
```diff
@@ -13,7 +13,7 @@ format:
 
 .PHONY: test
 test:
-	python3 -m tox -p auto
+	python3 -m tox
```
Comment (fredriv) [inline Makefile:16]: Use `-p auto -o` instead?

File: Makefile (hunk 1 above)
Comment (fredriv) [reply Makefile]: Ah, I see that was the old commit, never mind 😅

File: .github/workflows/continous-integration.yml [hunk 3]
```diff
@@ -1,24 +1,23 @@
...
   push:
     branches: master
 jobs:
-  build:
-
+  test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.6, 3.7, 3.8]
```
Comment (fredriv) [inline .github/workflows/continous-integration.yml:12]: Will this run all the tests (`flake8`, `black`, `mypy`) on all three Python versions? Or is this just ensuring that all three Python versions are available to `tox`?

File: Makefile (hunk 2 above)
Comment (fredriv) [inline Makefile:16]: Since we're running `tox` directly instead of `make test` in Github actions, revert this to run tests in parallel locally?

Comment (simenheg) [reply Makefile:16]: Then the output is garbled again. Or if we run in parallel without `-o`, warning messages are swallowed, and you can't use `breakpoint` for debugging tests, for instance.

File: .github/workflows/continous-integration.yml (hunk 3 above)
Comment (eide) [reply .github/workflows/continous-integration.yml:12]: It runs all the tests on all three versions, `tox -e py,flake8,black,mypy`.

We could run just `tox -e py` on each version to just run the tests against each python version, and then run flake8, black and mypy on just one python version if we wanted.

File: Makefile (hunk 2 above)
Comment (eide) [reply Makefile:16]: I don't have strong opinions here, as I usually just run `tox` directly myself. Often just running one env, ie `tox -e py`.

Comment (eide) [reply Makefile]: Remved this change, it's not related to the rest of the changes in this PR :ok_person:

Comment (fredriv) [reply Makefile]: I'm fine with adding `-o` here for improved output - I think we're doing that in all the other projects too 👍

**Review by fredriv — APPROVED** [review]

**Pull request merged**